relays/substrate-client/src/client.rs

```diff
@@ -100,15 +106,40 @@ where
 	}
 
 	/// Get a Substrate header by its number.
-	pub async fn header_by_number(&self, block_number: C::BlockNumber) -> Result<C::Header> {
+	pub async fn header_by_number(&self, block_number: C::BlockNumber) -> Result<C::Header>
+	where
+		C::Header: DeserializeOwned,
+	{
 		let block_hash = Self::block_hash_by_number(self, block_number).await?;
 		Ok(Self::header_by_hash(self, block_hash).await?)
 	}
 
+	/// Return runtime version.
+	pub async fn runtime_version(&self) -> Result<RuntimeVersion> {
+		Ok(Substrate::<C, _, _>::runtime_version(&self.client).await?)
+	}
+
+	/// Return native tokens balance of the account.
+	///
+	/// May panic if called on the chain that is not storing `frame_balances::AccountData`
```

Comment:
It doesn't seem it will panic though. I think it will just return garbage data, no?

Reply:
It calls `account_data_storage_key`, which may panic

Reply:
Irrelevant

Reply:
I still don't understand how this would panic, and I suppose it might, since the comment is still here. Can you explain which of the function calls can panic?

Reply:
Yeah, it should have been removed with `ChainWithBalances`.